### PR TITLE
[CRW 2.2] Update descriptions for devfiles

### PR DESCRIPTION
**Pull request by svor**
Signed-off-by: svor <vsvydenk@redhat.com>

### What does this PR do?
Updates the descriptions for devfiles by adding versions of frameworks and tools and adds missing tags.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-521

### CODE REVIEW

File: dependencies/che-devfile-registry/devfiles/02_java-jboss-fuse/meta.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 displayName: Red Hat Fuse
-description: Red Hat Fuse
-tags: ["OpenJDK", "Maven", "EAP"]
+description: Red Hat Fuse stack with EAP 7.2.7, OpenJDK 1.8 and Maven 3.5.4
```
Comment (nickboldt) [inline dependencies/che-devfile-registry/devfiles/02_java-jboss-fuse/meta.yaml:3]: Bearing in mind we plan to move to EAP 7.3 in CRW 2.2 ( https://issues.redhat.com/browse/CRW-340 ) this is accurate today.

File: dependencies/che-devfile-registry/devfiles/02_java-maven/meta.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 displayName: Java Maven
-description: Java stack with OpenJDK 8, Maven 3.5.4 and Vert.x demo application
+description: RHEL 8 Java stack with OpenJDK 8, Maven 3.5.4 and Vert.x demo application
```
Comment (nickboldt) [inline dependencies/che-devfile-registry/devfiles/02_java-maven/meta.yaml:3]: All the stacks are based on RHEL8. Why only this one gets the description??

File: dependencies/che-devfile-registry/devfiles/04_nodejs-mongo/meta.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
-displayName: NodeJS MongoDB Web Application
-description: Stack with NodeJS 10 and MongoDB 3.4
-tags: ["NodeJS", "Express", "MongoDB", "RealWorld", "ubi8", "Centos"]
+displayName: NodeJS MongoDB
+description: RHEL 8 NodeJS stack with NPM 6.13.4, NodeJS 10.19.0 and MongoDB 3.4
```
Comment (nickboldt) [inline dependencies/che-devfile-registry/devfiles/04_nodejs-mongo/meta.yaml:3]: Adding the z-stream digit is more accurate than excluding it... but also means we have to update this more often.

Do we want the extra churn? Can we also include a test in the rhel.Dockerfile that builds the devfile reg that verifies that the version in the container installed via RPM matches the version in the description? Or maybe just extract the version from `rpm -qa | grep npm` and use that value to insert here at build time?

File: dependencies/che-devfile-registry/devfiles/05_dotnet/meta.yaml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ---
 displayName: ".NET"
-description: .NET 3.1 stack with .NET Core SDK, Runtime, C# Language Support and Debugger
-tags: [".NET", "C#"]
+description: .NET stack with .NET Core SDK 3.1.103, Runtime, C# Language Support and Debugger
+tags: [".NET", "C#", ".NET SDK", ".NET Runtime", "Netcoredbg", "Omnisharp"]
```
Comment (nickboldt) [inline dependencies/che-devfile-registry/devfiles/05_dotnet/meta.yaml:4]: why no UBI8 tag here?

File: dependencies/che-devfile-registry/devfiles/05_go/meta.yaml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 ---
 displayName: Go
-description: Stack with Go 1.11.5
-tags: ["Go"]
+description: Stack with Go 1.12.12
+tags: ["Go", "Golang", "RHEL8"]
```
Comment (nickboldt) [inline dependencies/che-devfile-registry/devfiles/05_go/meta.yaml:4]: why RHEL8 and not UBI8?

File: dependencies/che-devfile-registry/devfiles/05_php-cake/meta.yaml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 ---
-displayName: "CakePHP Example"
-description: PHP Stack with PHP 7.1 and a quickstart CakePHP application for OpenShift v3
-tags: ["PHP", "Apache", "RHEL"]
+displayName: "PHP CakePHP"
+description: PHP Stack with PHP 7.3.5, Apache Web Server 2.4.37, Composer 1.8.4 and a quickstart CakePHP application for OpenShift v3
```
Comment (nickboldt) [inline dependencies/che-devfile-registry/devfiles/05_php-cake/meta.yaml:3]: does this application not work on OpenShift v4?  Has anyone tried it?

File: dependencies/che-devfile-registry/devfiles/05_php-web-simple/meta.yaml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 ---
 displayName: "PHP Simple"
-description: PHP Stack with PHP 7.1 and simple web application
-tags: ["PHP", "Apache", "RHEL"]
+description: PHP Stack with PHP 7.3.5, Apache Web Server 2.4.37 and Composer 1.8.4
+tags: ["PHP", "Apache", "Composer", "UBI8"]
```
Comment (nickboldt) [inline dependencies/che-devfile-registry/devfiles/05_php-web-simple/meta.yaml:4]: I see here we switched from RHEL to UBI8... but in other places we have NO UBI tag, or a RHEL one. Can we be consistent across ALL the devfiles please?

File: dependencies/che-devfile-registry/devfiles/02_java-jboss-fuse/meta.yaml (hunk 1 above)
Comment (svor) [reply dependencies/che-devfile-registry/devfiles/02_java-jboss-fuse/meta.yaml:3]: Yep, but for now it is still 7.2. We can update version in the descriptions when the issue will be done. I commented about it in the issue https://issues.redhat.com/browse/CRW-340?focusedCommentId=14038060&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-14038060

File: dependencies/che-devfile-registry/devfiles/05_go/meta.yaml (hunk 5 above)
Comment (svor) [reply dependencies/che-devfile-registry/devfiles/05_go/meta.yaml:4]: I checked it in the Dockerfile http://pkgs.devel.redhat.com/cgit/containers/codeready-workspaces-stacks-golang/tree/Dockerfile?h=crw-2.2-rhel-8#n13  and it extends `rhel8/go-toolset:1.12.8-45`

File: dependencies/che-devfile-registry/devfiles/02_java-maven/meta.yaml (hunk 2 above)
Comment (svor) [reply dependencies/che-devfile-registry/devfiles/02_java-maven/meta.yaml]: I've updated the description, now it doesn't contain information about RHEL. Now that information should be in `tags`

File: dependencies/che-devfile-registry/devfiles/04_nodejs-mongo/meta.yaml (hunk 3 above)
Comment (svor) [reply dependencies/che-devfile-registry/devfiles/04_nodejs-mongo/meta.yaml]: it is a good idea to automate this process but i don't think it should be done in this PR

File: dependencies/che-devfile-registry/devfiles/05_dotnet/meta.yaml (hunk 4 above)
Comment (svor) [reply dependencies/che-devfile-registry/devfiles/05_dotnet/meta.yaml]: Added

File: dependencies/che-devfile-registry/devfiles/05_php-cake/meta.yaml (hunk 6 above)
Comment (svor) [reply dependencies/che-devfile-registry/devfiles/05_php-cake/meta.yaml]: I've just checked it and it works, so I removed **v3** from the description

File: dependencies/che-devfile-registry/devfiles/05_php-web-simple/meta.yaml (hunk 7 above)
Comment (svor) [reply dependencies/che-devfile-registry/devfiles/05_php-web-simple/meta.yaml:4]: Now each stack has tag **UBI8** or **RHEL8** it depends which image was extended in the Dockerfile

**Pull request merged**